conda/shell/etc/fish/conf.d/conda.fish

```diff
@@ -109,6 +109,11 @@ function __fish_conda_commands
   string replace -r '.*_([a-z]+)\.py$' '$1' $_CONDA_ROOT/lib/python*/site-packages/conda/cli/main_*.py
   echo activate
   echo deactivate
+  echo env
```

Comment:
I think instead of `echo env` we should do the more general
```fish
  string replace -r '^.*/conda-' '' "$_CONDA_ROOT"/bin/conda-*
```
or maybe better
```fish
  for f in "$_CONDA_ROOT"/bin/conda-*
    if test -x "$f" -a ! -d "$f"
      string replace -r '^.*/conda-' '' "$f"
    end
  end
```
That way we would also pick up other commands, e.g., from `conda-build`.
NOTE: I have next to no `fish` experience, so I don't know certainly if the code snippet above is fine or not.

Reply:
Good idea! I think the first is fine. We don't have conda directories AFAIK, and we might discover other things to exclude later anyway.

Reply:
I was more thinking about the `test -x` part, i.e., to only pick up executables. The `-d` part is only to a necessary addition.
If one would do something like writing a (of course non-executable) `conda-build.log` file to `bin` or another stupidity, then we can easily exclude that with the second code snippet.
(BTW, `conda` itself doesn't do that executable test yet and thus will ungracefully exit if you were to run it with such `conda conda-build.log` -- I'll submit a PR for this.)

Reply:
Looks strange to me to have non-executable files in `bin` but ok, done. :)

Reply:
Definitely agree to that. But who knows what people might do :wink: -- we can always remove that check later on.
One case where we have non-executables would be on Windows where (due to no shebang support) a single `conda` script is replaced by `conda.exe` and `conda-script.py` (with the former essentially running `python conda-script.py`).
But well, to satisfy the "immensely large" Windows+`fish`+`conda` user base, we'd have to glob for `conda-*(.exe)?` or something :laughing:

Reply:
Hahaha this immensely large user base will certainly submit a PR if there's an issue :laughing: